queues/nats/nats_test.go

```diff
@@ -12,8 +14,11 @@ import (
 )
 
 func TestTransport(t *testing.T) {
-	transport := New()
-	vicetest.Transport(t, transport)
+	new := func() vice.Transport {
+		return New()
+	}
+
+	vicetest.Transport(t, new)
```

Comment:
Oh no, you're right. That's a shame. It's ok.

The implementations do need to return their concrete types, because they usually have specific stuff on them.